File: java/simulation/common/Node.java
package simulation.common;

public class Node {

private static int nodeCounter = 0;
public int nodeId;
public float r;
public int theta;
public String object;
public int attribute;
public boolean collision = false;
public boolean boundary = false;

public transient int x;
public transient int y;

public Node(){
	nodeId = nodeCounter;
	nodeCounter++;
}

public void setBoundaryTrue(){
	boundary = true;
}

public void setCollisionTrue(){
	collision = true;
}

public void setRTheta(float r, int theta){
	this.r = r;
	this.theta = theta;
}

public void defaultNode(){
	r = 0;
	theta = 0;
	object = "defaultNode";
	attribute = 0;
}

public String toString(){
	return "Node ID : " + nodeId + " Object : " + object + " x : " + x + " y : " + y ; 
}

public int compareTo(Node b) {
	if(this.attribute < b.attribute){
		return -1;
	}
	return 1;
}

}

File: src/simulation/common/Node.ts
let nodeCounter = 0;

export class Node {
  public nodeId: number;
  public r: number = 0;
  public theta: number = 0;
  public object: string = '';
  public attribute: number = 0;
  public collision: boolean = false;
  public boundary: boolean = false;
  public x: number = 0;
  public y: number = 0;

  constructor() {
    this.nodeId = nodeCounter;
    nodeCounter++;
  }

  setBoundaryTrue(): void {
    this.boundary = true;
  }

  setCollisionTrue(): void {
    this.collision = true;
  }

  setRTheta(r: number, theta: number): void {
    this.r = r;
    this.theta = theta;
  }

  defaultNode(): void {
    this.r = 0;
    this.theta = 0;
    this.object = 'defaultNode';
    this.attribute = 0;
  }

  toString(): string {
    return 'Node ID : ' + this.nodeId + ' Object : ' + this.object + ' x : ' + this.x + ' y : ' + this.y;
  }

  compareTo(b: Node): number {
    if (this.attribute < b.attribute) {
      return -1;
    }
    if (this.attribute === b.attribute) {
      return 0;
    }
    return 1;
  }
}
